### PR TITLE
hotfix for prior-link-geo peer review items

### PR DESCRIPTION
**Pull request by husobee**
### Summary

hotfix for prior-link-geo peer review items

### Type of change ( select one )

- [ ] Product feature
- [x] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: services/wallet/service.go [hunk 1]
```diff
@@ -433,31 +433,30 @@ func (service *Service) LinkWallet(
...
 				// then pass back the original geo error
 				return err
 			}
 			// allow invalid country if there was a prior linking
 		} else {
-			// if so ignore the region not supported error
-			return err
+			// not a invalid country error
+			// there was an unexpected error
+			return handlers.WrapError(err,
```
Comment (clD11) [inline services/wallet/service.go:455]: Does returning the wrapped errors affect the controller call i.e. will we be wrapping twice

Comment (husobee) [reply services/wallet/service.go]: addressed: https://github.com/brave-intl/bat-go/pull/1671/commits/9fd3136c1f3d30e39741617bbb05110368f56a6a

File: services/wallet/service.go [hunk 2]
```diff
@@ -433,31 +433,30 @@ func (service *Service) LinkWallet(
...
-			ok, err := service.Datastore.HasPriorLinking(
+			ok, priorErr := service.Datastore.HasPriorLinking(
 				ctx, infoID, uuid.NewV5(ClaimNamespace, userID))
-			if err != nil {
-				return fmt.Errorf("failed to check prior linkings: %w", err)
+			if priorErr != nil {
+				return fmt.Errorf("failed to check prior linkings: %w", priorErr)
 			}
 			if !ok {
 				// then pass back the original geo error
```
[the remark this answers is not in the document]
Comment (husobee) [reply services/wallet/service.go:450]: addressed: https://github.com/brave-intl/bat-go/pull/1671/commits/9fd3136c1f3d30e39741617bbb05110368f56a6a

File: services/wallet/service.go [hunk 3]
```diff
@@ -433,31 +433,30 @@ func (service *Service) LinkWallet(
...
 				return err
 			}
 			// allow invalid country if there was a prior linking
 		} else {
-			// if so ignore the region not supported error
-			return err
+			// not a invalid country error
+			// there was an unexpected error
```
[the remark this answers is not in the document]
Comment (husobee) [reply services/wallet/service.go]: addressed: https://github.com/brave-intl/bat-go/pull/1671/commits/9fd3136c1f3d30e39741617bbb05110368f56a6a

File: services/wallet/service.go [hunk 4]
```diff
@@ -433,31 +433,30 @@ func (service *Service) LinkWallet(
 	// verify that the user is kyc from uphold. (for all wallet provider cases)
 	if uID, ok, c, err := wallet.IsUserKYC(ctx, transactionInfo.Destination); err != nil {
 		// get the rewards wallet id from the uphold wallet info
-		infoID, err := uuid.FromString(info.ID)
-		if err != nil {
-			return fmt.Errorf("failed to parse wallet id: %w", err)
+		infoID, infoIDErr := uuid.FromString(info.ID)
+		if infoIDErr != nil {
+			return fmt.Errorf("failed to parse uphold id: %w", err)
```
[the remark this answers is not in the document]
Comment (husobee) [reply services/wallet/service.go]: addressed: https://github.com/brave-intl/bat-go/pull/1671/commits/9fd3136c1f3d30e39741617bbb05110368f56a6a

File: services/wallet/service.go [hunk 5]
```diff
@@ -370,9 +370,9 @@ func (service *Service) LinkGeminiWallet(ctx context.Context, walletID uuid.UUID
 	if err != nil {
 		// check if this gemini accountID has already been linked to this wallet,
 		if errors.Is(err, errorutils.ErrInvalidCountry) {
-			ok, err := service.Datastore.HasPriorLinking(
+			ok, priorErr := service.Datastore.HasPriorLinking(
 				ctx, walletID, uuid.NewV5(ClaimNamespace, accountID))
-			if err != nil {
+			if priorErr != nil {
 				return fmt.Errorf("failed to check prior linkings: %w", err)
```
Comment (clD11) [inline services/wallet/service.go:376]: should this be `priorErr`

**Review by clD11 — APPROVED** [review]

**Pull request merged**